CMakeLists.txt

```diff
@@ -39,6 +39,16 @@ if(WITH_TBB)
     add_definitions(-DWITH_TBB)
 endif()
 
+# OpenMP support
+set(WITH_OMP OFF CACHE BOOL "use OpenMP for on-node threading" )
```

Comment:
I think that it is reasonable to default to `serial`, and then throw an error if the user provides an unsupported/unknown back end. Silently just using serial if an invalid back end is requested will cause confusion.